x-pack/plugins/upgrade_assistant/__jest__/client_integration/helpers/time_manipulation.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+
+/**
+ * These helpers are intended to be used in conjunction with jest.useFakeTimers().
+ */
+
+const flushPromiseJobQueue = async () => {
+  // See https://stackoverflow.com/questions/52177631/jest-timer-and-promise-dont-work-well-settimeout-and-async-function
+  await Promise.resolve();
+};
+
+export const advanceTime = async (ms: number) => {
+  await act(async () => {
+    jest.advanceTimersByTime(ms);
+    await flushPromiseJobQueue();
```

Comment:
I think the flakiness in https://github.com/elastic/kibana/issues/111255 was due to not flushing this promise queue.